include/deal.II/lac/schur_complement.h

```diff
@@ -0,0 +1,498 @@
+    Range_2  &tmp_g = *(vector_memory_g.alloc());
+    Domain_1 &tmp_x = *(vector_memory_x.alloc());
+
+    // Reinitialise in context of how they'll be used
+    B.reinit_range_vector(tmp_f, /*bool omit_zeroing_entries =*/ true);
+    A_inv.reinit_range_vector(tmp_x, /*bool omit_zeroing_entries =*/ true);
+    C.reinit_range_vector(tmp_g, /*bool omit_zeroing_entries =*/ true);
+
+    // Need to form dst_g such that dst_g = S*src_y = (D - C*A_inv*B) src_y
+    if (D.is_null_operator == false)
+      D.vmult_add (dst_g, src_y); // dst_g += D*src_y (length y)
+
+    B.vmult (tmp_f, src_y); // tmp_f = B*src_y (length x)
+    try
+      {
+        A_inv.vmult (tmp_x, tmp_f); // tmp_x = A_inv*B*src_y (length x)
+      }
+    catch (...)
+      {
+        AssertThrow(false,
+                    ExcMessage("No convergence in A_inv vmult operation"));
+      }
+    C.vmult (tmp_g, tmp_x); // tmp_g = C*A_inv*B*src_y (length y)
+    dst_g -= tmp_g; // dst_g += D*src_y - C*A_inv*B*src_y
+
+    vector_memory_x.free(&tmp_x);
+    vector_memory_g.free(&tmp_g);
+    vector_memory_f.free(&tmp_f);
+  };
+
+  const auto vmult_add = return_op.vmult_add;
+  return_op.vmult = [vmult_add](Range_2 &dst_g, const Domain_2 &src_y)
+  {
+    dst_g = 0.;
+    vmult_add(dst_g, src_y);
+  };
+
+  return_op.Tvmult_add = [A_inv,B,C,D](Domain_2 &dst_g, const Range_2 &src_y)
+  {
+    static GrowingVectorMemory<Domain_1> vector_memory_f;
+    static GrowingVectorMemory<Domain_2> vector_memory_g;
+    static GrowingVectorMemory<Range_1>  vector_memory_x;
+
+    Domain_1 &tmp_f = *(vector_memory_f.alloc());
+    Domain_2 &tmp_g = *(vector_memory_g.alloc());
+    Range_1  &tmp_x = *(vector_memory_x.alloc());
+
+    // Reinitialise in context of how they'll be used
+    C.reinit_domain_vector(tmp_f, /*bool omit_zeroing_entries =*/ true);
+    A_inv.reinit_domain_vector(tmp_x, /*bool omit_zeroing_entries =*/ true);
+    B.reinit_domain_vector(tmp_g, /*bool omit_zeroing_entries =*/ true);
+
+    // Need to form y such that dst such that dst_g = S*src_y = (D^T - B^T*A_inv^T*C^T) src_y
+    if (D.is_null_operator == false)
+      D.Tvmult_add (dst_g, src_y); // dst_g += D^T*src_y (length y)
+
+    C.Tvmult (tmp_f, src_y); // tmp_f = C^T*src_y (length x)
+    try
+      {
+        A_inv.Tvmult (tmp_x, tmp_f); // tmp_x = A_inv^T*C^T*src_y (length x)
+      }
+    catch (...)
+      {
+        AssertThrow(false,
+                    ExcMessage("No convergence in A_inv Tvmult operation"));
+      }
+    B.Tvmult (tmp_g, tmp_x); // tmp_g = B^T*A_inv^T*C^T*src_y (length y)
+    dst_g -= tmp_g; // dst_g += D^T*src_y - B^T*A_inv^T*C^T*src_y
+
+    vector_memory_x.free(&tmp_x);
+    vector_memory_g.free(&tmp_g);
+    vector_memory_f.free(&tmp_f);
+  };
+
+  const auto Tvmult_add = return_op.Tvmult_add;
+  return_op.Tvmult = [Tvmult_add](Domain_2 &dst_g, const Range_2 &src_y)
+  {
+    dst_g = 0.;
+    Tvmult_add(dst_g, src_y);
+  };
+
+  return return_op;
+}
+
```

Comment:
I didn't check the implementation too closely but I assume that has been taken care of by others... ;-)